crates/ruff_linter/src/rules/pyflakes/rules/strings.rs

```diff
@@ -833,7 +881,17 @@ pub(crate) fn string_dot_format_extra_positional_arguments(
                 checker.locator(),
                 checker.stylist(),
             )?;
-            Ok(Fix::safe_edit(edit))
+            Ok(Fix::applicable_edit(
+                edit,
+                // Mark fix as unsafe if the call arguments contains a call with side effect
+                if call.arguments.arguments_source_order().any(|arg| {
```

Comment:
Same here, we should only make this fix unsafe if any removed argument has a side effect. It probably makes sense to add a test showing that the fix doesn't get marked as unsafe if any other (still used) argument has side effects.